File: artifacts/optipng-0.7.8/decompose/src/gz_header.c
typedef struct gz_header_s
{
  int text;
  uLong time;
  int xflags;
  int os;
  Bytef *extra;
  uInt extra_len;
  uInt extra_max;
  Bytef *name;
  uInt name_max;
  Bytef *comment;
  uInt comm_max;
  int hcrc;
  int done;
} gz_header
/*** DEPENDENCIES:
typedef unsigned long uLong
----------------------------
typedef Byte Bytef
----------------------------
typedef unsigned int uInt
----------------------------
typedef unsigned char Byte
----------------------------
***/

File: artifacts/lodepng/translate_with_test/src/gz_header.rs
use crate::*;

#[derive(Clone)]
pub struct GzHeader {
    pub text: i32,
    pub time: u64,  // uLong is u64
    pub xflags: i32,
    pub os: i32,
    pub extra: Option<Vec<u8>>,  // Bytef is u8
    pub extra_len: u32,  // uInt is u32
    pub extra_max: u32,  // uInt is u32
    pub name: Option<Vec<u8>>,  // Bytef is u8
    pub name_max: u32,  // uInt is u32
    pub comment: Option<Vec<u8>>,  // Bytef is u8
    pub comm_max: u32,  // uInt is u32
    pub hcrc: i32,
    pub done: i32,
}

